src/aap_eda/services/activation/activation_manager.py

```diff
@@ -356,18 +351,24 @@ def _is_already_stopped(self) -> bool:
             or self._is_in_status(ActivationStatus.ERROR)
         )
 
-    def _is_unresponsive(self) -> bool:
-        if self.db_instance.status in [
+    def _is_unresponsive(self, check_readiness: bool) -> bool:
+        previous_time = None
+        if check_readiness:
+            previous_time = self.latest_instance.started_at
+            timeout = settings.RULEBOOK_READINESS_TIMEOUT_SECONDS
+        elif self.db_instance.status in [
             ActivationStatus.RUNNING,
             ActivationStatus.STARTING,
         ]:
-            cutoff_time = timezone.now() - timedelta(
-                seconds=settings.RULEBOOK_LIVENESS_TIMEOUT_SECONDS,
-            )
-            return self.latest_instance.updated_at < cutoff_time
+            previous_time = self.latest_instance.updated_at
```

Comment:
@bzwei Can this instance be updated in the DB for any other reason other than the arrival of the session stats? I mean if some other code were to update the instance would we declare that the rulebook is running. Should we check if the session_stats is present and not empty.

Reply:
Right now I think is not the case, but I think it is a very valid concern, I agree to enhance this check.

Reply:
there is no other place that the updated_at is set. I will enhance the code to use current time just in case the session stat does not have the timestamp.